### PR TITLE
Add test to check dependent reads

### PR DESCRIPTION
**Pull request by Alan-Jowett**
Test to check behavior of dependent reads.

1) Check that packet size is > 4 bytes.
2) If true, set flag.
3) If flag is set, access packet.

```
alanjo@alanjo-dev2:~/ebpf-verifier$ sudo ./check ebpf-samples/build/dependent_read.o --dom linux
1,0.000427,4996
alanjo@alanjo-dev2:~/ebpf-verifier$ sudo ./check ebpf-samples/build/dependent_read.o
0,0.008814,5024
```

Verification of this sample passes on Linux verifier, but fails with prevail.

### CODE REVIEW

**Comment by Alan-Jowett** [comment]
Need to account for different packet layout on Linux vs Windows.

File: src/dependent_read.c [hunk 1]
```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Microsoft Corporation
+// SPDX-License-Identifier: MIT
+
+#include "bpf.h"
+
+__attribute__((section("xdp"), used)) void
+dependent_read(struct xdp_md* ctx)
+{
+    // Set r0 to 0 avoid returning uninitialized register.
```
Comment (dthaler) [inline src/dependent_read.c:9]:
```suggestion
    // Set r0 to 0 to avoid returning an uninitialized register.
```

File: src/dependent_read.c [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    asm volatile("r7 = %0" : : "r"(ctx->data));
+    // Load data_end into r8
+    asm volatile("r8 = %0" : : "r"(ctx->data_end));
+    // Compute packet length and store in r8.
+    asm volatile("r8 -= r7");
+    // Set flag register to 0
+    asm volatile("r9 = 0x0");
+    // Check if packet length (r8) is less than 4
+    asm volatile("if r8 < 4 goto l1");
+    // Set flag register to 1 (only hit if r8 >= 4.
```
Comment (dthaler) [inline src/dependent_read.c:22]:
```suggestion
    // Set flag register to 1 (only hit if r8 >= 4).
```

File: src/dependent_read.c [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+dependent_read(struct xdp_md* ctx)
+{
+    // Set r0 to 0 avoid returning uninitialized register.
+    asm volatile("r0 = 0");
+
+    // Load data into r7
+    asm volatile("r7 = %0" : : "r"(ctx->data));
+    // Load data_end into r8
+    asm volatile("r8 = %0" : : "r"(ctx->data_end));
+    // Compute packet length and store in r8.
```
Comment (dthaler) [inline src/dependent_read.c:16]: Good that this comment ends in punctuation but other lines don't.
Would be good to be consistent.

**Comment by Alan-Jowett** [comment]
Working on getting a version that passes Linux verifier, but fails prevail.

**Pull request merged**